### PR TITLE
Add GetSymbolUsesAtLocation API.

### PR DESCRIPTION
**Pull request by nojaf**
As discussed on the F# Foundation Slack, there are instances when the `GetSymbolUseAtLocation` function may return multiple correct symbols for a given line and end column. However, it currently only returns the first symbol in a somewhat arbitrary manner.

To address this limitation, I believe it would be beneficial to introduce an API that allows developers to filter out the symbol they specifically require, based on custom logic tailored to their use case.

In order to demonstrate the scenarios where this issue can arise, I have included two tests that highlight such situations. At present, the only available alternative is to retrieve all symbols, which can be quite inconvenient, particularly for larger files.

### CODE REVIEW

**Comment by T-Gro** [comment]
Now you can also change the single-case `GetSymbolUseAtLocation` to call into your new method, or?

File: src/Compiler/Service/FSharpCheckerResults.fs [hunk 1]
```diff
@@ -2089,6 +2089,36 @@ type internal TypeCheckInfo
                 Trace.TraceInformation(sprintf "FCS: recovering from error in GetSymbolUseAtLocation: '%s'" msg)
                 None)
 
+    member _.GetSymbolUsesAtLocation(line, lineStr, colAtEndOfNames, names) =
```
Comment (T-Gro) [inline src/Compiler/Service/FSharpCheckerResults.fs:2092]: Can you also de-duplicate thsi with the member right above?
If I did not overlook it, it is again the same thing (just differing in either list or option as return value)

**Review by T-Gro — APPROVED** [review]

File: src/Compiler/Service/FSharpCheckerResults.fs [hunk 2]
```diff
@@ -2089,6 +2089,36 @@ type internal TypeCheckInfo
...
+
+                match declItemsOpt with
+                | None -> List.empty
+                | Some (items, denv, _, m) ->
+                    items
+                    |> List.map (fun item ->
+                        let symbol = FSharpSymbol.Create(cenv, item.Item)
+                        symbol, item.ItemWithInst, denv, m))
+            (fun msg ->
+                Trace.TraceInformation(sprintf "FCS: recovering from error in GetSymbolUsesAtLocation: '%s'" msg)
```
Comment (psfinaki) [inline src/Compiler/Service/FSharpCheckerResults.fs:2119]: Any idea when this can actually happen? Just curious.

Comment (nojaf) [reply src/Compiler/Service/FSharpCheckerResults.fs:2119]: Not really, the old code had it. I guess it is a guard against something unexpected like a null pointer or something.

**Pull request merged**